### ISSUE TITLE
usdMtlx fails to read a mtlx file with more than one bind input

### ISSUE THREAD

**Issue by juliencb2018** [PixarAnimationStudios/USD#950]
### Description of Issue
usdMtlx is failing to properly read a materialX file that has more than one bind input.

### Steps to Reproduce
1. Input mtlx file with more than one bind input:

MultiBindInputs.mtlx

```
<?xml version="1.0"?>
<materialx version="1.36">                                                                                                                                                                                    
  <nodedef name="ND_layerShader" type="surfaceshader" node="layerShader">                                                                                                                                     
        <input name="weight_1" type="color3"/>                                                                                                                                                                
        <input name="weight_2" type="color3"/>                                                                                                                                                                
        <input name="weight_3" type="color3"/>                                                                                                                                                                
    </nodedef>                                                                                                                                                                                                
  <Gradient name="layered_layer1_gradient" type="color3">                                                                                                                                                     
  </Gradient>                                                                                                                                                                                                 
  <Gradient name="layered_layer2_gradient" type="color3">                                                                                                                                                     
  </Gradient>                                                                                                                                                                                                 
  <Gradient name="layered_layer3_gradient" type="color3">                                                                                                                                                     
  </Gradient>                                                                                                                                                                                                 
  <material name="layered">                                                                                                                                                                                   
    <shaderref name="layered_sr" node="layerShader">                                                                                                                                                          
      <bindinput name="weight_1" type="color3" output="o_layered_layer1_gradient" />                                                                                                                          
      <bindinput name="weight_2" type="color3" output="o_layered_layer2_gradient" />                                                                                                                          
      <bindinput name="weight_3" type="color3" output="o_layered_layer3_gradient" />                                                                                                                          
    </shaderref>                                                                                                                                                                                              
  </material>                                                                                                                                                                                                 
  <output name="o_layered_layer1_gradient" type="color3" nodename="layered_layer1_gradient" />                                                                                                                
  <output name="o_layered_layer2_gradient" type="color3" nodename="layered_layer2_gradient" />                                                                                                                
  <output name="o_layered_layer3_gradient" type="color3" nodename="layered_layer3_gradient" />                                                                                                                
</materialx>
```

2. python  calls to read and parse this file:

```
from pxr import Usd, UsdShade, Sdf
stage = Usd.Stage.CreateInMemory()
stage = stage.Open('MultiBindInputs.mtlx')
nodeGraph = UsdShade.NodeGraph.Get(stage, Sdf.Path('/MaterialX/Materials/layered/ND_layerShader'))
inputs = nodeGraph.GetInputs()
for i in inputs:
     print i.GetFullName(), i.HasConnectedSource(), i.GetConnectedSource()[0].GetPath()
```

3. The previous snippet does produce this warning:

```
Warning: in AddReference at line 1070 of px//lib/vendor/pixar/usdMtlx/pxr/usd/usdMtlx/reader.cpp -- Can't create node graph at </MaterialX/Materials/layered/NodeGraphs>; a '' already exists
Warning: in AddReference at line 1070 of pxr//lib/vendor/pixar/usdMtlx/pxr/usd/usdMtlx/reader.cpp -- Can't create node graph at </MaterialX/Materials/layered/NodeGraphs>; a '' already exists
```

And the prints are:
```
inputs:weight_1 True /MaterialX/Materials/layered/NodeGraphs/layered_layer1_gradient
inputs:weight_2 True /MaterialX/Materials/layered
inputs:weight_3 True /MaterialX/Materials/layered
```

which are incorrect.

The correct prints are:
```
inputs:weight_1 True /MaterialX/Materials/layered/NodeGraphs/layered_layer1_gradient
inputs:weight_2 True /MaterialX/Materials/layered/NodeGraphs/layered_layer2_gradient
inputs:weight_3 True /MaterialX/Materials/layered/NodeGraphs/layered_layer3_gradient
```
                                                                                                                                                                                                

### System Information (OS, Hardware)
Linux RHEL7

**Comment by jilliene** [comment]
Filed as internal issue #USD-5517

**Comment by spiffmon** [comment]
Thanks, @juliencb2018 - we'll have a look!

**Comment by juliencb2018** [comment]
Of course! I already have a fix and confidence test, I just need to
understand how we can access git and follow the procedures behind our
firewall.

--

Julien COHEN BENGIO | R&D | ILM SF | 415-746-2037

**Comment by sunyab** [comment]
PR #956 was released with 19.11 -- closing, thanks!
